src/linux/mod.rs

```diff
@@ -22,6 +22,7 @@ pub struct LinuxSandbox {
     env_exceptions: Vec<String>,
     landlock: RulesetCreated,
     allow_networking: bool,
+    full_env: bool,
```

Comment:
I'd suggest to merge `full_env` and `env_exceptions` into a single field:
```rust
enum EnvExceptions {
  Full,
  Partial(Vec<String>)
}
```
This way, the improper state where `full_env = true` and `!env_exceptions.is_empty()` isn't allowed to exist. The tradeoff is a bit more verbosity. This could be lifted outside of the OS-specific crates.

Reply:
On that note. Should either (or both) of these code snippets result in an error?

```
sandbox.add_exception(Exception::Environment("PATH".into()))?;
sandbox.add_exception(Exception::FullEnvironment)?;
```

```
sandbox.add_exception(Exception::FullEnvironment)?;
sandbox.add_exception(Exception::Environment("PATH".into()))?;
```

I don't really have strong opinions on it, but it might be worth documenting and testing those cases (especially if we decide that they should error).

Reply:
I'd say the state transition should be such that:
- the base case is `Partial(vec![])`, or a third variant, `None`.
- if I have `Full`, I can not transition to `Partial` and will error out if I try to.
- if I have `Partial` (and/, I can transition to `Partial` or to `Full`.

So the first snippet is ok, the second is an error.

Reply:
I mean that's a perfectly valid state? Exceptions are added one at a time for everything else, so pushing an entire vec for env seems inconsistent?